app/models/nic/interface.rb

```diff
@@ -67,18 +67,17 @@ def normalize_name
       # Remove whitespace
       self.name.gsub!(/\s/,'')
 
-      if domain.nil? and name.match(/\./)
+      if domain.nil? && name.match(/\./) && !changed_attributes['domain_id'].present?
         # try to assign the domain automatically based on our existing domains from the host FQDN
         self.domain = Domain.all.select{|d| name.match(/#{d.name}\Z/)}.first rescue nil
```

Comment:
ah yes, I already figured out and forgot, everytime I see this code I start to think if we can't move it to SQL :-) dismiss this review :-)